test/helpers/assertions.go

```diff
@@ -211,6 +212,47 @@ func (a *Assertions) PodDoesNotExist(podName string, msgAndArgs ...interface{})
 	a.podReachesState(podName, stateFunc, msgAndArgs...)
 }
 
+// Asserts that a Pod is running.
+func (a *Assertions) PodIsRunning(podName string, msgAndArgs ...interface{}) {
```

Comment:
@cheesesashimi can you please review this assertion to verify I am not missing something here. I added this after your initial review.

Reply:
Here's a few ideas for improvements:

```golang
func (a *Assertions) PodIsRunning(podName string, msgAndArgs ...interface{}) {
    // First, examine the current pod phase.
    isPodPhaseRunning := func(pod *corev1.Pod) bool {
        if pod == nil {
            return false
        }

        return pod.Status.Phasej == corev1.PodRunning
    }

    // Next, examine the individual containers on the pod.
    isPodContainersRunning := func(pod *corev1.Pod) bool {
        if pod == nil {
            return false
        }

        for _, container := range pod.Status.Containers {
            if container.State.Running == nil {
                return false
            }

            if container.State.Waiting != nil {
                return false
            }

            if container.State.Terminated != nil {
                return false
            }
        }

        return true
    }

    // Now, lets tie it all together.
    stateFunc := func(pod *corev1.Pod, err error) (bool, error) {
        return isPodPhaseRunning(pod) && isPodContainersRunning(pod), a.created(err)
    }

    a.podReachesState(podName, stateFunc, msgAndArgs...)
}
```